pandas/tests/tools/test_to_datetime.py

```diff
@@ -469,6 +469,89 @@ def test_to_datetime_mixed_datetime_and_string(self):
         expected = to_datetime([d1, d2]).tz_convert(pytz.FixedOffset(-60))
         tm.assert_index_equal(res, expected)
 
+    @pytest.mark.parametrize(
+        "fmt",
+        ["%Y-%d-%m %H:%M:%S%z", "%Y-%m-%d %H:%M:%S%z"],
+        ids=["non-ISO8601 format", "ISO8601 format"],
+    )
+    @pytest.mark.parametrize(
+        "utc, input, expected",
+        [
+            pytest.param(
+                True,
+                ["2000-01-01 01:00:00-08:00", "2000-01-01 02:00:00-08:00"],
+                DatetimeIndex(
+                    ["2000-01-01 09:00:00+00:00", "2000-01-01 10:00:00+00:00"],
+                    dtype="datetime64[ns, UTC]",
+                ),
+                id="all tz-aware, with utc",
+            ),
+            pytest.param(
+                False,
+                ["2000-01-01 01:00:00-08:00", "2000-01-01 02:00:00-08:00"],
+                DatetimeIndex(
+                    ["2000-01-01 01:00:00-08:00", "2000-01-01 02:00:00-08:00"],
+                    tz=pytz.FixedOffset(-480),
+                ),
+                id="all tz-aware, without utc",
+            ),
+            pytest.param(
+                True,
+                ["2000-01-01 01:00:00-08:00", "2000-01-01 02:00:00+00:00"],
+                DatetimeIndex(
+                    ["2000-01-01 09:00:00+00:00", "2000-01-01 02:00:00+00:00"],
+                    dtype="datetime64[ns, UTC]",
+                ),
+                id="all tz-aware, mixed offsets, with utc",
+            ),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "constructor",
+        [Timestamp, lambda x: Timestamp(x).to_pydatetime()],
```

Comment:
Nit: Could `datetime.fromisoformat` be used instead of `lambda x: Timestamp(x).to_pydatetime()`?

Reply:
yup - good one, thanks!

Reply:
This changes how the timezone is stored (timedelta vs pytz), unless it's UTC:
```python
In [10]: str1 = '2000-01-01 01:00:00-08:00'

In [11]: str2 = '2000-01-01 01:00:00+00:00'

In [12]: Timestamp(str1)
Out[12]: Timestamp('2000-01-01 01:00:00-0800', tz='pytz.FixedOffset(-480)')

In [13]: dt.datetime.fromisoformat(str1)
Out[13]: datetime.datetime(2000, 1, 1, 1, 0, tzinfo=datetime.timezone(datetime.timedelta(days=-1, seconds=57600)))

In [14]: Timestamp(str2)
Out[14]: Timestamp('2000-01-01 01:00:00+0000', tz='UTC')

In [15]: dt.datetime.fromisoformat(str2)
Out[15]: datetime.datetime(2000, 1, 1, 1, 0, tzinfo=datetime.timezone.utc)
```

It could work to have the strings be `'+00:00'`, but for the sake of testing I think it'd be better to have a non-UTC timezone-aware string - I'll revert to `to_pydatetime` for now then if that's OK

Reply:
Ah gotcha. Yeah using `to_pydatetime` is fine then